### PR TITLE
[GOBBLIN-1051] Emit Helix Leader Metrics

### PR DESCRIPTION
**Pull request by linweihs**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1051

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Test locally

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java [hunk 1]
```diff
@@ -58,15 +35,30 @@
 import com.linkedin.restli.server.resources.BaseResource;
 import com.typesafe.config.Config;
 import com.typesafe.config.ConfigFactory;
-
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
```
Comment (arjun4084346) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java]: this is not correct. please follow https://gobblin.readthedocs.io/en/latest/developer-guide/CodingStyle/ and use gobblin's codestyle

Comment (linweihs) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java]: import order restored by code style

**Comment by arjun4084346** [comment]
LGTM +1

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java [hunk 2]
```diff
@@ -25,6 +28,9 @@
 import java.util.Properties;
 import java.util.concurrent.TimeUnit;
 
+import org.slf4j.Logger;
```
Comment (sv2000) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java]: Are these unused imports? Don't see them being used in this PR?

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java [hunk 3]
```diff
@@ -645,4 +678,21 @@ private static void testGobblinService(GobblinServiceManager gobblinServiceManag
...
+  private static class HelixLeaderState {
+    private LeaderState state = LeaderState.UNKNOWN;
+  }
+
+  private enum LeaderState {
+    UNKNOWN(-1),
+    SLAVE(0),
+    MASTER(1);
+
+    public int value;
```
Comment (sv2000) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java]: Annotate with @Getter?

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java [hunk 4]
```diff
@@ -214,6 +223,14 @@ public GobblinServiceManager(String serviceName, String serviceId, Config config
       this.isGitConfigMonitorEnabled = false;
     }
 
+    // Initialize Helix leader guage
+    helixLeaderGauges = Optional.of(new HelixLeaderState());
+    String helixLeaderStateGaugeName =
+        MetricRegistry.name(MetricReportUtils.GOBBLIN_SERVICE_METRICS_PREFIX, ServiceMetricNames.HELIX_LEADER_STATE);
+    ContextAwareGauge<Integer> gauge = metricContext.newContextAwareGauge(helixLeaderStateGaugeName, () -> helixLeaderGauges.get().state.value);
```
Comment (sv2000) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java]: Use helixLeaderGauges.get().state.getValue() with Getter on value field?

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java (hunk 2 above)
Comment (linweihs) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java]: no. these are used on here 
`private static final Logger LOGGER = LoggerFactory.getLogger(GobblinServiceManager.class);`

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java (hunk 3 above)
Comment (linweihs) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java]: good point. added

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java (hunk 4 above)
Comment (linweihs) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/core/GobblinServiceManager.java]: updated.

**Pull request closed without merging**